src/orchestrator.ts

```diff
@@ -105,7 +104,10 @@ export class Orchestrator {
                 }
 
                 if ((partialResult instanceof Task || partialResult instanceof TaskSet) && partialResult.isFaulted) {
-                    g = gen.throw(partialResult.exception);
+                    if (!gen.throw) {
+                        throw new Error("Cannot properly throw the execption returned by customer code");
```

Comment:
What does this error mean? Why would `gen.throw` not be defined? Maybe we could add a comment explaining this?

Reply:
So TypeScript defines gen.throw as an optional method. In the wonderful world of JavaScript, it is completely possible to have methods be undefined, so we have to check for it to make the null-checker happy. I'll have to look more into what scenarios `gen.throw` would actually be undefined, so that I can write a more descriptive error message.

Reply:
So TypeScript defines gen.throw as an optional method. In the wonderful world of JavaScript, it is completely possible to have methods be undefined, so we have to check for it to make the null-checker happy. I'll have to look more into what scenarios `gen.throw` would actually be undefined, so that I can write a more descriptive error message.